knack/config.py

```diff
@@ -111,12 +111,12 @@ def items(self, section):
         import re
         pattern = self.env_var_name(section, '.+')
```

Comment:
Let's make `A-Z0-9` the only allows chars, so that we adhere to the doc https://docs.microsoft.com/en-us/cli/azure/azure-cli-configuration?view=azure-cli-latest#cli-configuration-values-and-environment-variables:

> Their corresponding environment variables are set as `AZURE_{section}_{name}`, in all caps.